kite-data/kite-data-core/src/main/java/org/kitesdk/data/Datasets.java

```diff
@@ -225,14 +336,38 @@
   /**
    * Update a {@link Dataset} for the given dataset or view URI.
    * <p>
-   * URI formats are defined by {@code Dataset} implementations, but must begin
-   * with "dataset:" or "view:".
+   * You can add columns, remove columns, or change the data type of columns 
+   * in your dataset, provided you don't attempt a change that would corrupt 
+   * the data. Kite follows the guidelines in the Avro schema. See 
+   * <a href="http://kitesdk.org/docs/current/Schema-Evolution.html">Schema
+   * Evolution</a>.
+   * <p>
+   * URIs must begin with {@code dataset:} or {@code view:}. The remainder of
+   * the URI is implementation specific, depending on the dataset scheme.
    *
-   * @param uri a {@code Dataset} or {@code View} URI.
-   * @param type the Java type of the entities in the dataset
-   * @param <E> The type of entities stored in the {@code Dataset}.
-   * @param <D> The type of {@code Dataset} expected.
-   * @return a newly created {@code Dataset} responsible for the given URI.
+   * @param uri a {@code Dataset} or {@code View} URI
+   * @param type a Java class that represents an entity in the dataset
+   * @param <E> the type used for readers and writers created by this {@code Dataset}
+   * @param <D> the type of {@code Dataset} expected
+   * @return a newly created {@code Dataset} responsible for the given URI
```

Comment:
Should also note that views are not returned by this method, even if a View URI is passed in.